core/variant/variant_utility.cpp

```diff
@@ -1234,6 +1234,108 @@ bool VariantUtilityFunctions::is_same(const Variant &p_a, const Variant &p_b) {
 	return p_a.identity_compare(p_b);
 }
 
+bool VariantUtilityFunctions::is_equal(const Variant &p_a, const Variant &p_b) {
+	const Variant::Type type_a = p_a.get_type();
+	const Variant::Type type_b = p_b.get_type();
+
+	if (type_a != type_b) {
```

Comment:
Should `1` (integer) and `1.0` (float) be equal?

Reply:
No, I don't think so. These are different numerical types.